src/main/java/com/labsynch/labseer/service/SaltServiceImpl.java

```diff
@@ -99,4 +169,369 @@ private void saveSalt(CmpdRegMolecule mol) throws IOException, CmpdRegMolFormatE
 		}
 	}
 
+	@Transactional
+	public String exportSalts() {
+		CmpdRegSDFWriter sdfWriter = null;
+		try {
+			sdfWriter = sdfWriterFactory.getCmpdRegSDFBufferWriter();
+		} catch (IllegalArgumentException e1) {
+			e1.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		// Get List of Salts 
+		List<Salt> allSaltList = Salt.findAllSalts();
+
+		for (Salt salt : allSaltList) {
+			try {
+				CmpdRegMolecule currSalt = cmpdRegMoleculeFactory.getCmpdRegMolecule(salt.getMolStructure());
+				currSalt.setProperty("name", salt.getName());
+				currSalt.setProperty("abbrev", salt.getAbbrev());
+				currSalt.setProperty("formula", salt.getFormula());
+				currSalt.setProperty("molWeight", String.valueOf(salt.getMolWeight()));
+				currSalt.setProperty("charge", String.valueOf(salt.getCharge()));
+				sdfWriter.writeMol(currSalt);
+			} catch (CmpdRegMolFormatException e) {
+				logger.error("bad structure error: " + salt.getMolStructure());
+			} catch (IOException e) {
+				logger.error("IO error reading in molfile");
+				e.printStackTrace();
+			}
+
+		}
+		try {
+			sdfWriter.close();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		logger.debug("ou	tput SDF: " + sdfWriter.getBufferString());
+		return sdfWriter.getBufferString();
+	}
+
+	public boolean isValidSaltEdit(Salt oldSalt, Salt newSalt)
```

Comment:
This function seems redundant with the much more fleshed out `validateSaltEdit` function. I think you're already doing these two checks within `validateSaltEdit`. Perhaps you could run `validateSaltEdit` then check whether it has any messages with level = `error` to derive your boolean "validSalt" ?